curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656706afd65547d22bee0b68.md

```diff
@@ -17,14 +17,9 @@ You should replace `pass` keyword with an `if` statement with this condition: `l
 
 ```js
 ({
-  test: () => {
-    const transformedCode = code.replace(/\r/g, "");
-    const merge_sort = __helpers.python.getDef("\n" + transformedCode, "merge_sort");
-    const { function_body } = merge_sort;
-
-    assert.match(function_body, /while\s+left_array_index\s*<\s*len\(\s*left_part\s*\)\s+and\s+right_array_index\s*<\s*len\(\s*right_part\s*\):\s*[^}]*if\s+left_part\s*\[\s*left_array_index\s*\]\s*<\s*right_part\s*\[\s*right_array_index\s*\]\s*:\s*pass/);
-  }
-})
+ test: () => assert(runPython(`
```

Comment:
this test is still passing for:

```py
while left_array_index < len(left_part) and right_array_index < len(right_part):
        pass
        if left_part[left_array_index] < right_part[right_array_index]:
            pass
```

The test can be modified to be run on the contents of `while`:

```js
({
 test: () => assert(runPython(`
  _Node(_code).find_function("merge_sort").find_whiles()[0].find_body().is_equivalent("if left_part[left_array_index] < right_part[right_array_index]:\\n  pass")`))
 })
```